connector/connector_ldap.go

```diff
@@ -126,15 +163,18 @@ func (cfg *LDAPConnectorConfig) Connector(ns url.URL, lf oidc.LoginFunc, tpls *t
 		}
 	}
 
-	if cfg.Timeout != 0 {
-		ldap.DefaultTimeout = cfg.Timeout * time.Millisecond
+	if cfg.Host == "" {
+		// For backward compatibility construct host form old fields.
+		cfg.Host = fmt.Sprintf("%s:%d", cfg.ServerHost, cfg.ServerPort)
```

Comment:
That's checked in the next line implicitly SplitHostPort. There previously wasn't a check if you didn't provide ServerHost, so I omitted one here. I'll add it back in though and be explicit.